kanalony-web-server/app/controllers/Application.scala

```diff
@@ -0,0 +1,126 @@
+package controllers
+
+import kanalony.storage.logic.queries.model._
+import kanalony.storage.logic._
+import model._
+import org.joda.time.DateTime
+import scala.concurrent.ExecutionContext.Implicits.global
+import model.Implicits._
+import play.api.mvc._
+import argonaut._
+import Argonaut._
+import scala.concurrent._
+
+class Application extends Controller {
+
+  def query: Action[AnyContent] = Action.async({ implicit request => doQuery(request) })
+
+  def doQuery(request : Request[AnyContent]) : Future[Result] = {
+    try {
+      val analyticsRequestOption = for {
+        input <- request.body.asJson
+        req <- input.toString.decodeOption[AnalyticsRequest]
+      } yield req
+
+      if (analyticsRequestOption.isEmpty) {
+        return wrapWithPromise(BadRequest("Invalid input - ensure proper JSON is supplied with the mandatory fields supplied and that the supplied MIME type is application/json"))
+      }
+
+      val queryParams = requestToQueryParams(analyticsRequestOption.get)
+      val queryExecutionResult = execute(queryParams)
+      val queryResponseData =  queryExecutionResult map { data => AnalyticsResponse(data.headers, data.rows) }
+
+      queryResponseData map {
+        data => Ok(data.asJson.toString)
```

Comment:
We need to discuss output format, as we'll need to support Json and CSV. If you are not handling this now, please add a JIRA ticket for supporting multiple output formats.